### PR TITLE
feat: read duplicate filtering and consensus reads using umi barcodes

### PR DESCRIPTION
**Pull request by jonca79**
### This PR:

(If this is a release PR, no need to add following. Leave this part empty)
(Use the following lines to create a PR text body. Make sure to remove all non-relevant one after you're done)
(Repeat each field as many times as necessary)

Added: for new features.
Changed: for changes in existing functionality.
Deprecated: for soon-to-be removed features.
Removed: for now removed features.
Fixed: for any bug fixes.
Security: in case of vulnerabilities.

### Review and tests:
- [ ] Tests pass
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Code review
- [ ] `CHANGELOG.md` is updated
- [ ] New code is executed and covered by tests, and test approve
- [ ] Module compatibility section in `README.md` is up to date

### CODE REVIEW

File: .tests/integration/reference/WASH7P.fna.dict
Comment (elleira) [inline .tests/integration/reference/WASH7P.fna.dict:1]: Should it really point to your home?

File: workflow/schemas/config.schema.yaml
Comment (elleira) [inline workflow/schemas/config.schema.yaml:1]: Mixing upper- and lower case in beginning on descriptions

Comment (jonca79) [reply workflow/schemas/config.schema.yaml:1]: Found only one place and then it was in complete sentences. You can't start a several sentences long description with a small letter. I refuse.

File: .tests/integration/reference/WASH7P.fna.dict
Comment (jonca79) [reply .tests/integration/reference/WASH7P.fna.dict:1]: changed

**Review by elleira — APPROVED** [review]

**Pull request merged**